### PR TITLE
First Draft of changes for BMC-SMM target phase-1

### PR DESCRIPTION
**Pull request by alexsmendez**

### CODE REVIEW

File: Exdi/exdigdbsrv/ExdiGdbSrv/LiveExdiGdbSrvServer.cpp [hunk 1]
```diff
@@ -1230,6 +1236,21 @@ HRESULT STDMETHODCALLTYPE CLiveExdiGdbSrvServer::GetContextEx(_In_ DWORD process
             pContext->DataSelector = static_cast<DWORD>(GdbSrvController::ParseRegisterValue32(registers["foseg"]));
         }
 
+        // Check if the GDT & IDT system registers
+        it = registers.find("gdtrbase");
+        if (it != registers.end())
```
Comment (PaeturPetersenMsft) [inline Exdi/exdigdbsrv/ExdiGdbSrv/LiveExdiGdbSrvServer.cpp:1241]: wouldn't the same code construct from above also be good here?
Like:
if (registers.find("gdtrbase") != registers.end())
{
   ...
}

File: Exdi/exdigdbsrv/ExdiGdbSrv/LiveExdiGdbSrvServer.cpp [hunk 2]
```diff
@@ -1230,6 +1236,21 @@ HRESULT STDMETHODCALLTYPE CLiveExdiGdbSrvServer::GetContextEx(_In_ DWORD process
...
 
+        // Check if the GDT & IDT system registers
+        it = registers.find("gdtrbase");
+        if (it != registers.end())
+        {
+            pContext->GDTBase = static_cast<DWORD>(GdbSrvController::ParseRegisterValue(registers["gdtrbase"]));
+            pContext->GDTLimit = static_cast<DWORD>(GdbSrvController::ParseRegisterValue(registers["gdtrlimit"]));
+        }
+
+        it = registers.find("idtrbase");
```
Comment (PaeturPetersenMsft) [inline Exdi/exdigdbsrv/ExdiGdbSrv/LiveExdiGdbSrvServer.cpp:1247]: Same as above?

File: Exdi/exdigdbsrv/ExdiGdbSrv/LiveExdiGdbSrvServer.cpp [hunk 3]
```diff
@@ -1189,7 +1189,14 @@ HRESULT STDMETHODCALLTYPE CLiveExdiGdbSrvServer::GetContextEx(_In_ DWORD process
         pContext->R13 = GdbSrvController::ParseRegisterValue(registers["r13"]);
         pContext->R14 = GdbSrvController::ParseRegisterValue(registers["r14"]);
         pContext->R15 = GdbSrvController::ParseRegisterValue(registers["r15"]);
-        pContext->EFlags = GdbSrvController::ParseRegisterValue32(registers["eflags"]);
+        if (registers.find("eflags") != registers.end())
+        {
+            pContext->EFlags = GdbSrvController::ParseRegisterValue32(registers["eflags"]);
+        }
+        else if (registers.find("rflags") != registers.end())
```
Comment (PaeturPetersenMsft) [inline Exdi/exdigdbsrv/ExdiGdbSrv/LiveExdiGdbSrvServer.cpp:1196]: So we only look for `rflags` if `eflags` are not present?

**Review by PaeturPetersenMsft — CHANGES_REQUESTED** [review]

**Review by PaeturPetersenMsft — APPROVED** [review]

**Comment by alexsmendez** [comment]
Integrated changes for BMC-SMM exdi gdbserver target

**Pull request merged**